From 83a9ca34b0f055ecef04549d004ca4efaf034e23 Mon Sep 17 00:00:00 2001
From: Arron Wang <arron.wang@intel.com>
Date: Tue, 30 Jul 2013 14:35:35 +0200
Subject: [PATCH 30/44] NFC: Fix secure element state check

Another typo from the initial commit where we check for the secure
element type field instead of its state when enabling or disabling it.

Signed-off-by: Arron Wang <arron.wang@intel.com>
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
(cherry picked from commit 225df91aba46a61c6adad8e0cf7fba048db3f7e4)
---
 net/nfc/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/nfc/core.c b/net/nfc/core.c
index d252912..ee1fe66 100644
--- a/net/nfc/core.c
+++ b/net/nfc/core.c
@@ -583,7 +583,7 @@ int nfc_enable_se(struct nfc_dev *dev, u32 se_idx)
 		goto error;
 	}
 
-	if (se->type == NFC_SE_ENABLED) {
+	if (se->state == NFC_SE_ENABLED) {
 		rc = -EALREADY;
 		goto error;
 	}
@@ -626,7 +626,7 @@ int nfc_disable_se(struct nfc_dev *dev, u32 se_idx)
 		goto error;
 	}
 
-	if (se->type == NFC_SE_DISABLED) {
+	if (se->state == NFC_SE_DISABLED) {
 		rc = -EALREADY;
 		goto error;
 	}
-- 
1.8.4.93.g57e4c17

